src/frontend/app/shared/components/start-end-date/start-end-date.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, OnInit, Output, EventEmitter, Input } from '@angular/core';
+import * as moment from 'moment';
+
+@Component({
+  selector: 'app-start-end-date',
+  templateUrl: './start-end-date.component.html',
+  styleUrls: ['./start-end-date.component.scss']
+})
+export class StartEndDateComponent {
+  @Output()
+  public endChange = new EventEmitter<moment.Moment>();
+  @Output()
+  public startChange = new EventEmitter<moment.Moment>();
+
+  @Output()
+  public isValid = new EventEmitter<boolean>();
+
+  get valid() {
+    return this.validValue;
+  }
+
+  set valid(valid: boolean) {
+    this.validValue = valid;
+    this.isValid.emit(this.validValue);
+  }
+
+  public validValue = true;
+
+  private startValue: moment.Moment;
+  private endValue: moment.Moment;
+
+
+  private isDifferentDate(oldDate: moment.Moment, newDate: moment.Moment) {
+    return !oldDate || !newDate || !oldDate.isSame(newDate);
+  }
+
+  private isStartEndValid(start: moment.Moment, end: moment.Moment) {
+    if (!end || !start) {
+      return true;
+    }
+    return start.isBefore(end);
+  }
+
+  @Input()
+  set start(start: moment.Moment) {
+    this.valid = true;
```

Comment:
There's a bug here, I'll raise in a separate issue. Once an invalid date range has been entered (start date after end date) the start date shows validation error. Changing the start date again to another invalid date removes the text. Note - the 'set' button is still correctly disabled.